packages/core/src/tracing/instrumentation/cloud/aws/sqs.js

```diff
@@ -321,6 +337,7 @@ function instrumentedReceiveMessage(ctx, originalReceiveMessage, originalArgs) {
         );
 
         promise.instanaAsyncContext = cls.getAsyncContext();
+        sqsConsumerAsyncContext = promise.instanaAsyncContext;
```

Comment:
Since this variable exists in the scope of our instrumentation module: What happens if two receive-messages calls happen at the same time? Or rather, if a second `receiveMessage` call is started while an earlier `receiveMessage` call is still in progress? Would we run the risk of overwriting the `sqsConsumerAsyncContext` of an ongoing call with a different async context?

In case this is actually a problem: Do we have access to the `promise` object in `instrumentedSQSConsumerExecuteHandler`? Or can we attach it to some object that we have access to in `instrumentedSQSConsumerExecuteHandler`?

Reply:
I thought about that too, but I assumed that this should not be a problem because:

1. Either this will run in a single thread or in a child process, which would then be a copy of the instrumentation (or am I wrong and they are shared anyway?)
2. The way SQS works is by long polling, which makes all messages to arrive together anyway, in the same context. There is no concurrent message processing, at least afaik.

But I will double check if this is a problem (it could be as I will have to fix tests).
I need to double check if we have access to the promise from executeHandler. From what I remember, we don't as sqs-consumer almost always return `await promise`... I need to go back to the code, as it may be that a lot of extra work needs to be done

Reply:
> Either this will run in a single thread or in a child process

I don't follow. AFAICT there is nothing in sqs-consumer that would guarantee (or even make it likely) that child processes or threads are used.  

> The way SQS works is by long polling

What if an app receives from multiple queues? Wouldn't that be multiple long polling requests that happen in parallel?

Reply:
I managed to implement a local reference to the async context in a new version.
Could you please take a look? Essentially, I patched `sqsConsumer.sqs.receiveMessage.promise` from the `receiveMessage` function from SQS-Consumer, and added promise.instanaAsyncContext to ctx (sqs this reference).
Not the global instana async context is gone